### PR TITLE
Fix save-analysis generation with extern_in_paths/extern_absolute_paths

### LINKED ISSUE #48742: save-analysis: inferred crates are not included
save-analysis includes crates declared with `extern crate` or in the implicit prelude, but does not include those inferred from imports due to the new `extern crate` elision feature. I assume because `extern_crate` on the tcx does not include info for those crates (see https://github.com/rust-lang/rust/blob/master/src/librustc_save_analysis/lib.rs#L113).

This is the root cause for part of https://github.com/rust-lang-nursery/rls/issues/729

### PR DESCRIPTION
**Pull request by sinkuu**
Fixes #48742.

### CODE REVIEW

File: src/librustc_metadata/creader.rs [hunk 1]
```diff
@@ -367,16 +367,31 @@ impl<'a> CrateLoader<'a> {
         let cmeta = self.cstore.get_crate_data(cnum);
         let mut old_extern_crate = cmeta.extern_crate.borrow_mut();
 
+        fn path_len_reverse(src: ExternCrateSource) -> cmp::Reverse<usize> {
+            cmp::Reverse(match src {
+                ExternCrateSource::Extern { path_len, .. } |
+                ExternCrateSource::Use { path_len } => path_len,
+                _ => usize::max_value(),
```
Comment (petrochenkov) [inline src/librustc_metadata/creader.rs]: It looks like it'd be more convenient to move `path_len` into `ExternCrate` and initialize it to `usize::max_value()` for `ExternCrateSource::Path`.
(Maybe some better heuristic than `max_value` could be found later).

**Comment by petrochenkov** [comment]
r=me after addressing the comment

File: src/librustc_metadata/creader.rs (hunk 1 above)
Comment (sinkuu) [reply src/librustc_metadata/creader.rs]: Done.

**Comment by sinkuu** [comment]
Fixed.

**Pull request merged**